### PR TITLE
Feature/DGI-46: Modal to edit accountant for account

### PR DESCRIPTION
**Pull request by ntw-ITM**
I have made a lot of changes in this branch that wasn't originally part of this issue, but I decided that they were necessary to make this feature work properly. This is why so many files have changed.

Some changes:
- I decided I had to remove Email as an index from the fluent configuration of Users, because if eg. you add an accountant with an email that has been added before, but has since been deactivated, it won't allow it. Because of this change, I made a new migration.
- There were some errors in CreateAccountantCommand in the way it checked if the provided email was already in use. I think this is also an error in some other commands, which I will check and maybe fix in another PR.
- As we talked about yesterday, it makes sense to use entity extensions. This has been implemented here as well, because it was needed for the feature.
- Because I'm starting to have deeply nested components in the table view of accounts that need to eg. fetch data, I have made an AccountsContext that is provided in components/accounts.tsx.
- I have made a few modifications to the table view of accounts to better display the accountant and option to edit it.

### CODE REVIEW

**Comment by ntw-ITM** [comment]
@objmj-itminds  and @CSCITM , I know that the backend build check fails. This is because GetAccountsQueryTest suddenly fails. I'm fairly sure it has something to do with the entity extension of Account and the mappings on AccountDto. But I don't understand what's wrong. Can you spot it?

**Comment by ntw-ITM** [comment]
Request for review retracted, as I've chosen to make more changes.

**Review by objmj-itminds — CHANGES_REQUESTED** [review]
Don't remove email but do as we discussed earlier regarding activating old users.

File: backend/Domain/EntityExtensions/AccountExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public static User GetClient(this Account account)
+    {
+      try
+      {
+        var user = account.Users.First(u => u.Role == RoleEnum.Client && u.DeactivationTime == null);
+        return user;
+      } catch
+      {
+        return null;
+      }
```
Comment (objmj-itminds) [inline backend/Domain/EntityExtensions/AccountExtensions.cs:18]: If you use `FirstOrDefault` there shouldn't be a need to try/catch.

File: frontend/src/components/Accounts/ChangeAccountant/CurrentAccountant.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  accountant: IUserAccountIdDto;
+}
+
+const CurrentAccountant: FC<Props> = ({ accountant }) => {
+  const { t } = useLocales();
+  const toast = useToast();
+  const { fetchData, isFetching } = useContext(AccountsContext);
+
+  const handleDelete = useCallback(
+    async (e: React.MouseEvent) => {
```
Comment (objmj-itminds) [inline frontend/src/components/Accounts/ChangeAccountant/CurrentAccountant.tsx:31]: In most cases, we never want the actual 'event'. If a parameter is always used in one singular way, then move that logic out of the callback.

**Review by objmj-itminds — APPROVED** [review]

**Pull request merged**